include/create/types.h

```diff
@@ -33,11 +33,41 @@ POSSIBILITY OF SUCH DAMAGE.
 #define CREATE_TYPES_H
 
 #include <vector>
+#include <stdint.h>
+#include <string>
+#include <stdexcept>
 
 namespace create {
-  enum RobotModel {
-    CREATE_1 = 0, // Roomba 400 series
-    CREATE_2      // Roomba 600 series
+  enum ProtocolVersion {
+    V_1 = 1,
+    V_2 = 2,
+    V_3 = 4,
+    V_ALL = 0xFFFFFFFF
+  };
+
+
+
+  class RobotModel {
+    public:
+      uint32_t id;
+      ProtocolVersion version;
+      float axleLength;
+      unsigned int baud;
+      float maxVelocity;
+      float wheelDiameter;
+
+      RobotModel(const std::string& name);
+
+      bool operator==(RobotModel& other);
+      operator uint32_t();
+
+      static RobotModel ROOMBA_400; // Roomba 400 series
```

Comment:
It would be nice if these were `const`, but I can't think of an elegant way to enforce it. The only thing I can think of is to make these private and provide getter functions for them.